sympy/solvers/tests/test_solveset.py

```diff
@@ -1661,6 +1661,12 @@ def test_nonlinsolve_polysys():
     system = [x**2 + y - 3, x - y - 4]
     assert nonlinsolve(system, (x, y)) != nonlinsolve(system, (y, x))
 
+    assert nonlinsolve([-x**2 - y**2 + z, -2*x, -2*y, 1], [x, y, z]) == S.EmptySet
```

Comment:
When I try this example on master I get:
```
AttributeError: 'int' object has no attribute 'is_Relational'
```
If this was being called somewhere then why didn't this error show up before? Is there code that catches AttributeError? (There shouldn't be)

Reply:
Should it also sympify the `symbols` argument?

Reply:
I was trying to figure out what the existing code is doing as regards sympifying the `symbols` argument, and I noticed some inconsistencies. `solve` sympifies the `symbols` argument so that something like `solve(x+1 'x')` returns `[-1]`. `solveset` also sympifies the symbols argument. However, `linsolve` does not symify its `symbols` argument and neither does `nonlinsolve` (on master). @oscarbenjamin do you see anything wrong with sympifying `symbols` in nonlinsolve. Since `solve` already does it, I don't think it should do any harm.

Reply:
All arguments should be sympified if they are expected to be Basic. The `_sympify` function should be used though and string input should not be accepted.